core/shared/src/main/scala/scodec/bits/ByteVector.scala

```diff
@@ -2262,7 +2262,8 @@ object ByteVector extends ByteVectorCompanionCrossPlatform {
     private val bvlen = bv.size.toInt
     private val pos = new CustomAtomicInteger(0)
 
-    override def read(): Int = bv.get(pos.getAndIncrement().toLong)
+    override def read(): Int =
+      if (pos.get() >= bvlen) -1 else bv.get(pos.getAndIncrement().toLong) & 0xff
```

Comment:
Safer to get current value of `pos` once instead of calling it twice here.

```scala
val p = pos.getAndIncrement()
if (p >= bvlen) -1
else bv.get(p.toLong) & 0xff
```

Reply:
I don't think it is a good idea, since `available` then will return negative values if `read()` is called at the end of the stream.

Reply:
Okay, my concern is with thread safety - a race between concurrent read calls and an interleaving of get/get/getAndIncrement/getAndOncrement.

Reply:
Ah I see. So a fix be in `available()` instead I guess.